Document new GQLSTATUS 22NB9 (#303)
neo4j/docs-status-codes: modules/ROOT/pages/errors/gql-errors/22NB9.adoc

= 22NB9

== Status description

error: data exception - invalid inner list type. Lists cannot have `{ <<item>> }` as an inner type in this context.

== Example scenario

> For example, when trying to create a property type constraint with a list of a union as property type:

Reviewer: ```suggestion
For example, try to create a property type constraint with a list of a union as property type:
```

[source,cypher]
----
CREATE CONSTRAINT myConstraint
FOR (n:Label)
REQUIRE n.prop IS :: LIST<INTEGER NOT NULL | FLOAT NOT NULL>
----

> The error will have GQLSTATUS xref:errors/gql-errors/50N11.adoc[50N11].

Reviewer: How about something like this:

```suggestion
You will receive the GQLSTATUS error code xref:errors/gql-errors/50N11.adoc[50N11], which is triggered by two causes:
```
```

Author: Not convinced, 'triggered by causes' is a good way to put it. And we need to be clear about the order of the cause chain it is 50N11 -> 22N90 -> 22NB9 and not e.g. 50N11 -> 22NB9 -> 22N90

Author: What do you think about something like this? Would it be easier to read?

You will receive an error with the GQLSTATUS xref:errors/gql-errors/50N11.adoc[50N11].
The error will have a cause with the GQLSTATUS xref:errors/gql-errors/22N90.adoc[22N90]. 
This cause will in turn have a cause with GQLSTATUS 22NB9 and status description:

Author: I could also add the status description for each of the three levels if that is helpful or type out the structure in more detail, just not sure if that make sense given that this page is about 22NB9 and not really 22N90 or 50N11

Reviewer: Yeah, exactly. It contains two causes, but here we're talking just about one of them.
Maybe something like:

You will receive an error with GQLSTATUS xref:errors/gql-errors/50N11.adoc[50N11].
This error has a cause detailed in xref:errors/gql-errors/22N90.adoc[22N90], which also has a subsequent cause with GQLSTATUS 22NB9 and status description:

> The first cause in the cause chain will have GQLSTATUS xref:errors/gql-errors/22N90.adoc[22N90] and the second cause will have GQLSTATUS 22NB9.

Reviewer: ```suggestion

* xref:errors/gql-errors/22N90.adoc[22N90]
```

> The status description of the cause with GQLSTATUS 22NB9 will be:

Author: I'm not sure this is the best way to phrase it, happy to take improvement advice

> The status description of the cause with GQLSTATUS 22NB9 will be:

Reviewer: ```suggestion
* The current 22NB9 with status description:
```

> 

Reviewer: ```suggestion
+
```

Reviewer: Yes, I meant an extra empty line here.
